flatten.js

```diff
@@ -1,16 +1,13 @@
 // Problem 3: Complete the flatten function that takes a JS Object, returns a JS Object in flatten format (compressed)
 function flatten(unflatObject) {
   var output = {};
-
-  for (var iteration in unflatObject) {
-    if (typeof unflatObject[iteration] == "object") {
-      var flattenObject = flatten(unflatObject[iteration]);
-
+  for (var index in unflatObject) {
+    if (typeof unflatObject[index] == "object") {
+      var flattenObject = flatten(unflatObject[index]);
       for (var count in flattenObject) {
-        output[iteration + "." + count] = flattenObject[count];
+        output[index + "." + count] = flattenObject[count];
       }
-    } else output[iteration] = unflatObject[iteration];
+    } else output[index] = unflatObject[index];
```

Comment:
It's good to encapsulate even a single line if, else, for in braces as it improved readability and if not that it definitely should always at least be on the next line with proper indentation.